sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiver.cs

```diff
@@ -305,18 +417,25 @@ await foreach (ServiceBusMessage msg in ret.ConfigureAwait(false))
         /// </summary>
         /// <param name="fromSequenceNumber"></param>
         /// <param name="maxMessages"></param>
-        /// <param name="sessionId"></param>
         /// <param name="cancellationToken"></param>
         /// <returns></returns>
         internal async IAsyncEnumerable<ServiceBusMessage> PeekRangeBySequenceInternal(
             long? fromSequenceNumber,
             int maxMessages = 1,
-            string sessionId = null,
             [EnumeratorCancellation]
             CancellationToken cancellationToken = default)
         {
-            string receiveLinkName = "";
+            if (IsSessionReceiver)
+            {
+                // if this is a session receiver, the receive link must be open in order to peek messages
+                await RetryPolicy.RunOperation(
+                    async () => await Consumer.ReceiveLink.GetOrCreateAsync(RetryPolicy.Options.TryTimeout).ConfigureAwait(false),
```

Comment:
The original intent was to calculate the try timeout to allow custom retry policies to provide logic that adjusts the timeout period based on the attempt.

Reply:
Fixed